src/handlers/ExecuteViewSubmitHandler.ts

```diff
@@ -421,4 +429,180 @@ export class ExecuteViewSubmitHandler {
+                );
+
+                if (validationError) {
+                    if (room) {
+                        await sendNotification(this.read, this.modify, user, room, {
+                            message: validationError
+                        });
+                    }
+
+                    return this.context.getInteractionResponder().errorResponse();
+                }
+            }
+
+            const userPreferenceStorage = new UserPreferenceStorage(
+                this.persistence,
+                this.read.getPersistenceReader(),
+                user.id,
+            );
+            const existingPreference = await userPreferenceStorage.getUserPreference();
+
+            const currentLLMConfig = existingPreference.llmConfiguration || {} as any;
+            const updatedPreference = {
+                ...existingPreference,
+                llmConfiguration: {
+                    ...currentLLMConfig,
+                    llmUsagePreference: llmUsagePreferenceValue || currentLLMConfig.llmUsagePreference || LLMUsagePreferenceEnum.Workspace,
+                    llmProvider: llmProviderValue || currentLLMConfig.llmProvider || LLMProviderEnum.Groq,
+                    ...(selfHostedUrlValue && { selfHosted: { url: selfHostedUrlValue } }),
+                    ...(openaiApiKeyValue && { openai: { apiKey: openaiApiKeyValue } }),
+                    ...(geminiApiKeyValue && { gemini: { apiKey: geminiApiKeyValue } }),
+                    ...(groqApiKeyValue && { groq: { apiKey: groqApiKeyValue } }),
+                    // Preserve existing API keys if no new values provided
+                    ...(!selfHostedUrlValue && currentLLMConfig.selfHosted && { selfHosted: currentLLMConfig.selfHosted }),
+                    ...(!openaiApiKeyValue && currentLLMConfig.openai && { openai: currentLLMConfig.openai }),
+                    ...(!geminiApiKeyValue && currentLLMConfig.gemini && { gemini: currentLLMConfig.gemini }),
+                    ...(!groqApiKeyValue && currentLLMConfig.groq && { groq: currentLLMConfig.groq }),
+                },
+            };
+
+            await userPreferenceStorage.storeUserPreference(updatedPreference);
+
+            if (room) {
+                await sendNotification(this.read, this.modify, user, room, {
+                    message: t(Translations.LLM_CONFIGURATION_SUCCESS, currentLanguage)
+                });
+            }
+
+            return this.context.getInteractionResponder().successResponse();
+
+        } catch (error) {
+            this.app.getLogger().error('Error saving LLM configuration:', error);
+
+            try {
+                if (room) {
+                    const errorLanguage = await getUserPreferredLanguage(
+                        this.read.getPersistenceReader(),
+                        this.persistence,
+                        user.id,
+                    );
+                    if (error.message?.includes('api') ||
```

Comment:
This is also a bad practice to many or cases and this can be easily handled by defining and array of string and use that

Reply:
you can also try to get some status code from the AI API for this you don't have to reply of messages